### DISCUSSION TITLE
Is it Possible to open 2 Windows?

### DISCUSSION THREAD

**Discussion by nschmidtdev** [saucer/saucer#37] [Q&A] [answered]
An error occurs when opening a second window. The message refers to thread safety or the inability to open CoreWebView2.  Therefore, the question arises: Is it even possible to open a second window?

**Comment by Curve** [comment]
Yes, it is possible, you simply cannot open a new webbiew from an asynchronous callback directly.

When you want to open a webview from another thread, you should use the `dispatch` method of `saucer::application` to make the creation happen on the main thread.

The tests do this. See here: https://github.com/saucer/saucer/blob/7e9c730adcb46f90e38e193c2135ae6ed621ec95/tests/include/test.hpp#L30

The `safe_delter` used there is also really important.

I may consider adding a utility which does this to `saucer::application` in a future version.

**Comment by Curve** [comment]
Also updated the docs regarding this: https://saucer.app/docs/common-pitfalls#construction-from-another-thread

**Comment by Curve** [comment, marked as answer]
This was greatly simplified with version 7.0.0, which now makes construction and cleanup completely thread-safe by default :)
